sablo/META-INF/resources/sablo/js/sablo_app.ts

```diff
@@ -297,7 +295,13 @@ angular.module('sabloApp', ['webSocketModule', 'webStorageModule']).value("$sabl
 		currentServiceCallDone = false
 		currentServiceCallWaiting = times.length
 		currentServiceCallTimeouts = times.map(function(t) { return setTimeout(callServiceCallbacksWhenDone, t) })
-		return promise.then(markServiceCallDone, markServiceCallFailed)
+		promise.then(markServiceCallDone, markServiceCallFailed);
```

Comment:
Before it was chaining the promise in deferredEvents and the one returned to the user. And so we could be sure that markServiceCallDone and markServiceCallFailed are called before the similar user code executes - which could end up calling another api - setting for example currentServiceCallDone again before it was reset; and then it would be cleared again. Do we have a guarantee that the functions assigned to then one after another get executed in the same order?

If we do then this code change is fine.
If we don't then we could revert it and maybe define on the "then-able" as you called it an Object.defineProperty that when requestInfo is set on that one it will also be set on the "promise" from deferredEvents.

Reply:
Thanks.